System/RTETools.cpp

```diff
@@ -107,4 +107,28 @@ namespace RTE {
 	bool WithinBox(Vector &point, Vector &boxPos, float width, float height) {
 		return point.m_X >= boxPos.m_X && point.m_X < (boxPos.m_X + width) && point.m_Y >= boxPos.m_Y && point.m_Y < (boxPos.m_Y + height);
 	}
-}
\ No newline at end of file
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	std::string RoundFloatToPrecision(float input, int precision, int roundingMode) {
+		if (roundingMode == 0) {
+			std::stringstream floatStream;
+			floatStream << std::fixed << std::setprecision(precision) << input;
+			return floatStream.str();
+		} else {
+			RTEAssert(powf(10.0F, static_cast<float>(precision)) < std::numeric_limits<float>::max(), "PRECISION SET GREATER THAN EXPONENT MAX!");
+			RTEAssert(powf(10.0F, static_cast<float>(precision)) > 0, "NEGATIVE PRECISION WILL YEILD DIVIDE BY ZERO ERROR!");
+			RTEAssert(input < (std::numeric_limits<float>::max() / powf(10.0F, static_cast<float>(precision))), "VALUE WILL EXCEED NUMERIC LIMITS WITH PRECISION " + std::to_string(precision));
+			if (roundingMode == 1) {
+				float roundingBuffer = floor(input * powf(10.0F, static_cast<float>(precision))) / powf(10.0F, static_cast<float>(precision));
+				return RoundFloatToPrecision(roundingBuffer, precision);
+			} else if (roundingMode == 2) {
+				float roundingBuffer = ceil(input * powf(10.0F, static_cast<float>(precision))) / powf(10.0F, static_cast<float>(precision));
+				return RoundFloatToPrecision(roundingBuffer, precision);
+			} else {
+				RTEAssert(false, "Error in RoundFloatToPrecision: INVALID ROUNDING MODE");
```

Comment:
You wanna use RTEAbort for this. No need for a return after it, and it's obviously cleaner

Reply:
Aye, that was just VS suggesting I avoid possible completion without return conditions.